src/components/Alert/Alert.js

```diff
@@ -19,26 +22,31 @@ class Alert extends React.Component {
     dismissible: PropTypes.bool.isRequired,
     idString: PropTypes.string,
     analyticsString: PropTypes.string,
-    children: PropTypes.node.isRequired
-  };
-
-  ref = React.createRef();
+    children: PropTypes.node.isRequired,
+    handleAlertDismissed: PropTypes.func
+  }
 
   get className() {
     const { type, className } = this.props
     const baseClass = sparkComponentClassName('Alert')
     const variantClass = sparkComponentClassName('Alert', null, type)
+    const displayClass = sparkClassName('utility', 'Display', null, 'none')
 
-    return classnames(baseClass, variantClass, { [className]: className })
+    return classnames(baseClass, variantClass, { [className]: className, [displayClass]: this.state && this.state.dismissed })
   }
 
-  componentDidMount = () => {
-    const { dismissible } = this.props
+  get classNameCloseIcon() {
+    const baseClass = sparkComponentClassName('Alert', 'icon')
+    const variantClass = sparkComponentClassName('Alert', 'icon', 'dismiss')
 
-    if (dismissible) {
-      alerts(this.ref.current, {})
-    }
-  };
+    classnames(baseClass, variantClass)
+  }
+
+  dismissAlert = () => {
+    this.setState({dismissed: true}, () => {
+      this.props.handleAlertDismissed && this.props.handleAlertDismissed(true)
```

Comment:
you can default handleAlertDismissed in default props to an empty function and there wont be a need to check for it's existence

Reply:
yes good call